src/main/java/net/java/otr4j/io/SerializationUtils.java

```diff
@@ -470,6 +471,20 @@ public static String byteArrayToHexString(byte in[]) {
 	}
 
 	/**
+     * Convert the {@code String} text to a {@code byte[]}, including sanitizing
+     * it to make sure no corrupt characters conflict with bytes that have
+     * special meaning in OTR. Mostly, this means removing NULL bytes, since
+     * {@code 0x00) is used as the separator between the message and the TLVs
+     * in an OTR Data Message.
+     *
+     * @param msg the plain text message being sent
+     * @return byte[] the incoming message converted to OTR-safe bytes
+     */
+    public static byte[] convertTextToBytes(String msg) {
+        return msg.replace('\0', '?').getBytes(SerializationUtils.UTF8);
```

Comment:
That is handled by the calling code:
```java
                if (msgText != null && msgText.length() > 0) {
                    try {
                        out.write(SerializationUtils.convertTextToBytes(msgText));
                    } catch (IOException e) {
                        throw new OtrException(e);
                    }
                }
```

And it makes sense to have the calling code handling it because `out.write(null)` will throw a `NullPointerException`[1](http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html#write(byte[],%20int,%20int)).